src/v2/blobs.rs

```diff
@@ -62,60 +47,32 @@ impl Client {
                 }
             }
         };
-        let req = match self.new_request(hyper::Method::GET, url.clone()) {
-            Ok(r) => r,
-            Err(e) => {
-                let msg = format!("new_request failed: {}", e);
-                error!("{}", msg);
-                return Box::new(futures::future::err::<_, _>(Error::from(msg)));
-            }
-        };
-        let freq = self.hclient.request(req);
-        let fres = freq
-            .from_err()
-            .inspect(move |_| {
-                trace!("GET {:?}", url);
-            }).and_then(move |r| {
-                match r.status() {
-                    StatusCode::MOVED_PERMANENTLY
-                    | StatusCode::TEMPORARY_REDIRECT
-                    | StatusCode::FOUND => {
-                        trace!("Got moved status {:?}", r.status());
-                    }
-                    _ => return Either::A(future::ok(r)),
-                };
-                let redirect: Option<String> = match r.headers().get("Location") {
-                    None => {
-                        return Either::A(future::err(Error::from(
-                            "get_blob: missing location header",
-                        )))
-                    }
-                    Some(loc) => {
-                        trace!("Got Location header {:?}", loc);
-                        String::from_utf8(loc.as_bytes().to_vec()).ok()
+
+        let fres = reqwest::async::Client::new()
+            .get(url)
+            .send()
+            .inspect(|res| trace!("Blob GET status: {:?}", res.status()))
+            .and_then(|mut res| {
+                let body =
+                    std::mem::replace(res.body_mut(), reqwest::async::Decoder::empty()).concat2();
```

Comment:
This looks a bit funky, but I'm not familiar enough with it to know if there is a more straightforward way to do this.

Reply:
I've tried to avoid the `std::mem::replace` but couldn't find a way to satisfy the borrow checker, The lifetime of `res` is not long enough, and it also can't be cloned. This seems to leave no other safe option as using `replace` here.
I found a document which explains it in more detail: https://github.com/rust-unofficial/patterns/blob/master/idioms/mem-replace.md